diff --git a/config.def.h b/config.def.h
index 86b7e76..48c7a2e 100644
--- a/config.def.h
+++ b/config.def.h
@@ -2,18 +2,24 @@ static const char *background_color = "#3e3e3e";
 static const char *border_color = "#ececec";
 static const char *font_color = "#ececec";
 static const char *font_pattern = "monospace:size=10";
-static const unsigned line_spacing = 5;
-static const unsigned int padding = 15;
+static unsigned line_spacing = 5;
+static unsigned int padding = 15;
 
-static const unsigned int width = 450;
-static const unsigned int border_size = 2;
-static const unsigned int pos_x = 30;
-static const unsigned int pos_y = 60;
+static unsigned int width = 450;
+static unsigned int border_size = 2;
+static unsigned int pos_x = 30;
+static unsigned int pos_y = 60;
+
+static unsigned int critical_notif = 0;
+static char *background_color_critical = "#442222";
+static char *border_color_critical = "#CC6666";
+static char *font_color_critical = "#CC6666";
 
 enum corners { TOP_LEFT, TOP_RIGHT, BOTTOM_LEFT, BOTTOM_RIGHT };
 enum corners corner = TOP_RIGHT;
 
-static const unsigned int duration = 5; /* in seconds */
+static unsigned int duration = 5; /* in seconds */
+static unsigned int duration_critical = 10; /* in seconds */
 
 #define DISMISS_BUTTON Button1
 #define ACTION_BUTTON Button3
diff --git a/herbe.c b/herbe.c
index 51d3990..2c533f9 100644
--- a/herbe.c
+++ b/herbe.c
@@ -1,5 +1,6 @@
 #include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>
+#include <X11/Xresource.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <signal.h>
@@ -15,6 +16,13 @@
 #define EXIT_FAIL 1
 #define EXIT_DISMISS 2
 
+#define XRES_STR(name)                                        \
+	if (XrmGetResource(db, "herbe." #name, "*", &type, &val)) \
+	name = val.addr
+#define XRES_INT(name)                                        \
+	if (XrmGetResource(db, "herbe." #name, "*", &type, &val)) \
+	name = strtoul(val.addr, 0, 10)
+
 Display *display;
 Window window;
 int exit_code = EXIT_DISMISS;
@@ -81,12 +89,18 @@ void expire(int sig)
 
 int main(int argc, char *argv[])
 {
+	int i = 0;
+
 	if (argc == 1)
 	{
 		sem_unlink("/herbe");
 		die("Usage: %s body", argv[0]);
 	}
 
+	for (i = 1; i < argc; i++)
+		if (!strcmp(argv[i], "-c"))  /* critical */
+			critical_notif = 1; 
+
 	struct sigaction act_expire, act_ignore;
 
 	act_expire.sa_handler = expire;
@@ -107,6 +121,33 @@ int main(int argc, char *argv[])
 	if (!(display = XOpenDisplay(0)))
 		die("Cannot open display");
 
+	XrmInitialize();
+
+	char *res_man = XResourceManagerString(display);
+	XrmDatabase db = XrmGetStringDatabase(res_man);
+
+	char *type;
+	XrmValue val;
+
+	XRES_STR(background_color);
+	XRES_STR(border_color);
+	XRES_STR(font_color);
+	XRES_STR(background_color_critical);
+	XRES_STR(border_color_critical);
+	XRES_STR(font_color_critical);
+	XRES_STR(font_pattern);
+
+	XRES_INT(line_spacing);
+	XRES_INT(padding);
+	XRES_INT(width);
+	XRES_INT(border_size);
+	XRES_INT(pos_x);
+	XRES_INT(pos_y);
+	XRES_INT(corner);
+
+	XRES_INT(duration);
+	XRES_INT(duration_critical);
+
 	int screen = DefaultScreen(display);
 	Visual *visual = DefaultVisual(display, screen);
 	Colormap colormap = DefaultColormap(display, screen);
@@ -117,6 +158,14 @@ int main(int argc, char *argv[])
 	XSetWindowAttributes attributes;
 	attributes.override_redirect = True;
 	XftColor color;
+
+	if (critical_notif) {
+		font_color = font_color_critical;
+		background_color = background_color_critical;
+		border_color = border_color_critical;
+		duration = duration_critical;
+	}
+
 	XftColorAllocName(display, visual, colormap, background_color, &color);
 	attributes.background_pixel = color.pixel;
 	XftColorAllocName(display, visual, colormap, border_color, &color);
@@ -133,6 +182,9 @@ int main(int argc, char *argv[])
 
 	for (int i = 1; i < argc; i++)
 	{
+		if (i==1 && !strcmp(argv[i], "-c"))
+			i++;
+
 		for (unsigned int eol = get_max_len(argv[i], font, max_text_width); eol; argv[i] += eol, num_of_lines++, eol = get_max_len(argv[i], font, max_text_width))
 		{
 			if (lines_size <= num_of_lines)
@@ -214,7 +266,8 @@ int main(int argc, char *argv[])
 	XftDrawDestroy(draw);
 	XftColorFree(display, visual, colormap, &color);
 	XftFontClose(display, font);
+	XrmDestroyDatabase(db);
 	XCloseDisplay(display);
 
 	return exit_code;
-}
\ No newline at end of file
+}
