diff --git a/config.def.h b/config.def.h
index 1c0b587..4b28003 100644
--- a/config.def.h
+++ b/config.def.h
@@ -28,7 +28,16 @@ static const Rule rules[] = {
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	{ "Firefox",  NULL,       NULL,       1 << 1,       0,           -1 },
+	{ NULL,    "scratchpad",  NULL,       0,            1,           -1 },
+	{ NULL,    "urxvtscrat",  NULL,       0,            1,           -1 },
+	{ "Putty",    NULL,       NULL,       0,            1,           -1 },
+	{ "Filezilla", NULL,      NULL,       1 << 4,       0,           -1 },
+	{ "com-aquafold-datastudio-DataStudio", NULL,      NULL,       1 << 3,       0,           -1 },
+	{ "Eclipse",  NULL,       NULL,       1 << 0,       0,           -1 },
+	{ "Firefox-esr", NULL, "Session Timeout - Mozilla Firefox",      1 << 8,       1,           -1 },
+	{ "floatterm",  NULL,     NULL,       0,            1,           -1 },
+	{ "stmusic",    NULL,     NULL,       0,            1,           -1 },
 };
 
 /* layout(s) */
@@ -44,7 +53,7 @@ static const Layout layouts[] = {
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -62,7 +71,7 @@ static const char *termcmd[]  = { "st", NULL };
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
-	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
+	{ MODKEY,                       XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
@@ -70,7 +79,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
 	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
 	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
-	{ MODKEY,                       XK_Return, zoom,           {0} },
+	{ MODKEY|ShiftMask,             XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
 	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
@@ -93,7 +102,18 @@ static Key keys[] = {
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	{ MODKEY,                       XK_ntilde, spawn,         SHCMD("~/bin/scratchpad") },
+	{ MODKEY|ShiftMask,             XK_ntilde, spawn,         SHCMD("~/bin/scratchpad urxvtscrat") },
+	{ MODKEY|ShiftMask,             XK_e,      spawn,         SHCMD("~/bin/eclipse_start.sh") },
+	{ MODKEY|ShiftMask,             XK_d,      spawn,         SHCMD("wmname LG3D;/opt/datastudio/datastudio.sh") },
+	{ MODKEY|ShiftMask,             XK_p,      spawn,         SHCMD("~/bin/dmenu/chooseputty") },
+	{ MODKEY|ShiftMask,             XK_m,      spawn,         SHCMD("~/bin/dmenu/choosemusic") },
+	{ MODKEY|ShiftMask,             XK_w,      spawn,         SHCMD("~/bin/wallpaper.sh") },
+	{ MODKEY|ShiftMask,             XK_plus,   spawn,         SHCMD("~/bin/audioscript up 5") },
+	{ MODKEY|ShiftMask,             XK_minus,  spawn,         SHCMD("~/bin/audioscript down 5") },
+	{ MODKEY|ShiftMask,             XK_KP_Add, spawn,         SHCMD("~/bin/audioscript up 5") },
+	{ MODKEY|ShiftMask,             XK_KP_Subtract, spawn,    SHCMD("~/bin/audioscript down 5") },
+	{ MODKEY|ShiftMask,             XK_y,      spawn,         SHCMD("~/bin/dmenu/yt -r") },
 };
 
 /* button definitions */
